python/ray/dag/tests/test_accelerated_dag.py

```diff
@@ -286,6 +294,54 @@ def f(x):
         dag.experimental_compile()
 
 
+def test_dag_fault_tolerance_chain(ray_start_regular_shared):
+    actors = [
+        Actor.remote(0, fail_after=100 if i == 0 else None, sys_exit=False)
+        for i in range(4)
+    ]
+    with InputNode() as i:
+        dag = i
+        for a in actors:
+            dag = a.echo.bind(dag)
+
+    compiled_dag = dag.experimental_compile()
+
+    for i in range(99):
+        output_channels = compiled_dag.execute(i)
+        # TODO(swang): Replace with fake ObjectRef.
+        results = output_channels.begin_read()
+        assert results == i
+        output_channels.end_read()
+
+    with pytest.raises(RuntimeError):
+        for i in range(99):
```

Comment:
Why loop 99 times? Shouldn't this already throw an exception on the first iteration because fail_after is set to 100?

Reply:
Ah the failures are randomized.